packages/service-library/src/servicelib/aiohttp/profiler_middleware.py

```diff
@@ -1,43 +1,46 @@
+import json
+
 from aiohttp.web import HTTPInternalServerError, Request, StreamResponse, middleware
 from pyinstrument import Profiler
 from servicelib.mimetype_constants import (
     MIMETYPE_APPLICATION_JSON,
     MIMETYPE_APPLICATION_ND_JSON,
 )
 
-from .._utils_profiling_middleware import append_profile
+from .._utils_profiling_middleware import _request_profiler, append_profile
 
 
 @middleware
 async def profiling_middleware(request: Request, handler):
-    profiler: Profiler | None = None
-    if request.headers.get("x-profile") is not None:
-        profiler = Profiler(async_mode="enabled")
-        profiler.start()
-
-    response = await handler(request)
-
-    if profiler is None:
-        return response
-    if response.content_type != MIMETYPE_APPLICATION_JSON:
-        raise HTTPInternalServerError(
-            reason=f"Profiling middleware is not compatible with {response.content_type=}",
-            headers={},
+    if profiler_kwargs := request.headers.get("x-profile"):
```

Comment:
so the "profiling" comes in via a header?
Do we also have a setting for enabling/disabling the profiler?

Reply:
This is done manually. This profiler is only for debugging. The profiling middleware is only enabled in master deployments. The way I usually use it is from the admin panels where I can send requests within the swarm. Then I do something like
```bash
 curl -X 'POST'   'http://10.43.103.193.nip.io:30006/v2/computations'   -H 'accept: application/json'   -H 'Content-Type: application/json'   -d '{
  "user_id": 1,
  "project_id": "33d145cc-44e0-4fa0-8c0e-446445837c53",
  "start_pipeline": true,
  "product_name": "osparc",
  "force_restart": false,
  "cluster_id": 0,
  "simcore_user_agent": "",
  "use_on_demand_clusters": false
}' -H 'x-profile: {"async_mode": "enabled"}' -v | jq -r .profile

```
to get a profile of that endpoint